staging/src/k8s.io/apiserver/pkg/server/options/serving.go

```diff
@@ -340,11 +341,13 @@ func CreateListener(network, addr string, config net.ListenConfig) (net.Listener
 		network = "tcp"
 	}
 
-	ln, err := config.Listen(context.TODO(), network, addr)
+	l, err := config.Listen(context.TODO(), network, addr)
 	if err != nil {
 		return nil, 0, fmt.Errorf("failed to listen on %v: %v", addr, err)
 	}
 
+	ln := &proxyproto.Listener{Listener: l}
```

Comment:
What if someone doesn't want/need the proxy. If there is a problem with this code it cannot just be turned off with a flag.

Reply:
Thank you. I will add an option to enable/disable this feature.